.github/workflows/ci.yml

```diff
@@ -0,0 +1,71 @@
+name: ci
+on:
+  push:
+    branches: [main, release]
+  pull_request:
+jobs:
+  prepare:
+    runs-on: ubuntu-latest
+    outputs:
+      web: ${{ steps.web.outputs.any_changed }}
+      server: ${{ steps.server.outputs.any_changed }}
+      worker: ${{ steps.worker.outputs.any_changed }}
+    steps:
+      - name: checkout
+        uses: actions/checkout@v4
+      - name: changed files for web
+        id: web
+        uses: tj-actions/changed-files@v36
+        with:
+          files: |
+            web
+            .github/workflows/ci.yml
+            .github/workflows/ci_web.yml
+
+      - name: changed files for server
+        id: server
+        uses: tj-actions/changed-files@v36
+        with:
+          files: |
+            server
+            .github/workflows/ci.yml
+            .github/workflows/ci_server.yml
+            .github/workflows/build_server.yml
+
+      - name: changed files for worker
+        id: worker
+        uses: tj-actions/changed-files@v36
+        with:
+          files: |
+            worker
+            .github/workflows/ci.yml
+            .github/workflows/ci_worker.yml
+            .github/workflows/build_worker.yml
+
+  ci-web:
+    name: web
+    needs: prepare
+    if: needs.prepare.outputs.web == 'true'
+    uses: ./.github/workflows/ci_web.yml
+
+  ci-server:
+    name: server
+    needs: prepare
+    if: needs.prepare.outputs.server == 'true'
+    uses: ./.github/workflows/ci_server.yml
+
+  ci-worker:
+    name: worker
+    needs: prepare
+    if: needs.prepare.outputs.worker == 'true'
+    uses: ./.github/workflows/ci_worker.yml
+
+  ci:
```

Comment:
Question ✋ 

May I ask what is this for?

Reply:
this is the required action to allow the merge